ktor-features/ktor-metrics-micrometer-prometheus/jvm/src/io/ktor/metrics/micrometer/prometheus/MicrometerMetricsPrometheus.kt

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright 2014-2019 JetBrains s.r.o and contributors. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package io.ktor.metrics.micrometer.prometheus
+
+import io.ktor.application.Application
+import io.ktor.application.ApplicationCall
+import io.ktor.application.call
+import io.ktor.application.feature
+import io.ktor.http.ContentType
+import io.ktor.http.charset
+import io.ktor.http.content.OutgoingContent
+import io.ktor.metrics.micrometer.AbstractMicrometerMetrics
+import io.ktor.response.ApplicationSendPipeline
+import io.ktor.response.respond
+import io.ktor.routing.get
+import io.ktor.routing.routing
+import io.ktor.util.cio.bufferedWriter
+import io.micrometer.core.instrument.distribution.DistributionStatisticConfig
+import io.micrometer.prometheus.PrometheusConfig
+import io.micrometer.prometheus.PrometheusMeterRegistry
+import io.prometheus.client.Collector
+import io.prometheus.client.exporter.common.TextFormat
+import kotlinx.coroutines.io.ByteWriteChannel
+import java.util.*
+
+class MicrometerMetricsPrometheus(config: Configuration) :
+    AbstractMicrometerMetrics<PrometheusMeterRegistry, MicrometerMetricsPrometheus.Configuration>(config) {
+
+    internal fun process(value: MetricsContent) = object : OutgoingContent.WriteChannelContent() {
+
+        override val contentType = ContentType.parse(TextFormat.CONTENT_TYPE_004)
+
+        override suspend fun writeTo(channel: ByteWriteChannel) {
+            val metricSamplesToScrape = value.filteredMetricFamilySamples()
+
+            channel.bufferedWriter(contentType.charset() ?: Charsets.UTF_8).use {
+                TextFormat.write004(it, metricSamplesToScrape)
+            }
+        }
+    }
+
+    class Configuration : AbstractMicrometerMetrics.AbstractConfiguration<PrometheusMeterRegistry>() {
+
+        /**
+         * The meter registry
+         */
+        override var registry = PrometheusMeterRegistry(PrometheusConfig.DEFAULT)
+
+        override var distributionStatisticConfig: DistributionStatisticConfig = DistributionStatisticConfig.Builder()
+            .percentilesHistogram(true)
+            .build()
+
+        /**
+         * Defines if a metrics endpoint is installed under "/metrics" for all metrics registered at [registry].
+         * Set to false if you want to declare a custom metric endpoint instead.
+         */
+        var installDefaultEndpoint = true
```

Comment:
Instead of the flag something like `defaultEndpointRoute: String? = "/metrics"` can be used so both enable/disable and path customization will be accomplished at once.